src/VisualStudio/Core/Def/DesignerAttribute/VisualStudioDesignerAttributeService.cs

```diff
@@ -40,28 +40,29 @@ internal class VisualStudioDesignerAttributeService
         /// </summary>
         private readonly IServiceProvider _serviceProvider;
 
-        /// <summary>
-        /// Our connection to the remote OOP server. Created on demand when we startup and then
-        /// kept around for the lifetime of this service.
-        /// </summary>
-        private RemoteServiceConnection<IRemoteDesignerAttributeDiscoveryService>? _lazyConnection;
-
         /// <summary>
         /// Cache from project to the CPS designer service for it.  Computed on demand (which
         /// requires using the UI thread), but then cached for all subsequent notifications about
         /// that project.
         /// </summary>
-        private readonly ConcurrentDictionary<ProjectId, IProjectItemDesignerTypeUpdateService?> _cpsProjects
-            = new();
+        private readonly ConcurrentDictionary<ProjectId, IProjectItemDesignerTypeUpdateService?> _cpsProjects = new();
 
         /// <summary>
         /// Cached designer service for notifying legacy projects about designer attributes.
         /// </summary>
         private IVSMDDesignerService? _legacyDesignerService;
 
-        // We'll get notifications from the OOP server about new attribute arguments. Batch those
-        // notifications up and deliver them to VS every second.
-        private readonly AsyncBatchingWorkQueue<DesignerAttributeData>? _workQueue;
+        /// <summary>
+        /// Cancellation series controlling the individual pieces of work added to <see cref="_workQueue"/>.  Every time
+        /// we add a new item, we cancel the prior item so that batch can stop as soon as possible and move onto the
+        /// next batch.
+        /// </summary>
+        private readonly CancellationSeries _cancellationSeries = new();
+        private readonly AsyncBatchingWorkQueue<CancellationToken> _workQueue;
```

Comment:
wonder if it would be worth creating another `AsyncBatchingWorkQueue` that internally makes a cancellation, that way we can just pick between the regular queue and the cancellation queue

Reply:
interesting.  let me look into that.  it would be nice for this common pattern.